lib/gringotts/gateways/sagepay.ex

```diff
@@ -296,13 +315,21 @@ defmodule Gringotts.Gateways.SagePay do
   # `respond`.
 
   # @spec commit(_) :: {:ok | :error, Response}
-  def commit(:post, endpoint, params, opts) do
+  defp commit(:post, endpoint, params, opts) do
     a_url = @url <> endpoint
+
+    HTTPoison.post(a_url, params, opts)
+    |> respond
+  end
+
+  defp commit_for_key(:post, endpoint, params, opts) do
```

Comment:
Since there is only 1 call site for this function, we can move the logic there.

Reply:
Since the logic of commit_for_key is totally different from the commit. So, I think we can't move.

Reply:
My bad! I didn't realise that `commit_for_key` is called from 2 functions. However, `format_response` can be moved inside `commit_for_key`.